src/search-headless.ts

```diff
@@ -555,5 +558,74 @@ export default class SearchHeadless {
   setFilterOption(filter: SelectableStaticFilter): void {
     this.stateManager.dispatchEvent('filters/setFilterOption', filter);
   }
+
+  /**
+   * Perform a generativeDirectAnswer request to the query most recent search stored in state.
+   *
+   * @returns A Promise of a {@link GenerativeDirectAnswerResponse} from the Search API or
+   *          of undefined if there is no results defined in state
+   */
+  async executeGenerativeDirectAnswer(): Promise<GenerativeDirectAnswerResponse | undefined> {
+    const searchId = this.state.meta.uuid;
+    if (!searchId) {
+      console.error('no search id supplied for generative direct answer');
+      return;
+    }
+    let requestName: 'universalQuery' | 'verticalQuery';
+    let results: VerticalResults[] | undefined;
+    if (this.state.meta.searchType === SearchTypeEnum.Vertical) {
+      requestName = 'verticalQuery';
+      if (this.state.vertical.appliedQueryFilters && this.state.vertical.queryDurationMillis 
```

Comment:
can you make a typescript type guard function for the universal/vertical results and invoke here? (https://www.typescriptlang.org/docs/handbook/advanced-types.html)

Reply:
I added the function in src/models/slices/vertical. Please let me know if it looks incorrect